dspace-api/src/main/java/org/dspace/app/sitemap/GenerateSitemaps.java

```diff
@@ -237,14 +244,37 @@ public static void generateSitemaps(boolean makeHTMLMap, boolean makeSitemapOrg)
 
         while (allItems.hasNext()) {
             Item i = allItems.next();
-            String url = uiURLStem + "/items/" + i.getID();
-            Date lastMod = i.getLastModified();
 
-            if (makeHTMLMap) {
-                html.addURL(url, lastMod);
-            }
-            if (makeSitemapOrg) {
-                sitemapsOrg.addURL(url, lastMod);
+            DiscoverQuery entityQuery = new DiscoverQuery();
```

Comment:
if for performance reason we need to make a Discovery query here why not use only discovery/solr to build the sitemaps avoiding the initial items findAll and the single solr query for each items?

Reply:
We kept the items findAll to ensure there's no risk of the sitemaps to be incomplete if the search index is being re-built
With the codebase in this PR, it ensures the items are still present, but simply use the /items/uuid syntax if the discovery index is incomplete due to a forced index during the sitemap generation

Reply:
ok, maybe we can create a backlog issue to further improve that in a future version? we could query the db for the number of items and solr, if they say the same we can use just solr as a reindex is not ongoing for sure.

Reply:
I went ahead and logged a ticket in the backlog to consider refactoring to use Discovery/Solr in the future: https://github.com/DSpace/DSpace/issues/3182   I agree though that it is low priority at this time.